src/storage/petastorm_storage_engine.py

```diff
@@ -0,0 +1,126 @@
+# coding=utf-8
+# Copyright 2018-2020 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from src.spark.session import Session
+from src.catalog.models.df_metadata import DataFrameMetadata
+from petastorm.etl.dataset_metadata import materialize_dataset
+from src.storage.abstract_storage_engine import AbstractStorageEngine
+from src.readers.petastorm_reader import PetastormReader
+from src.models.storage.batch import Batch
+
+from petastorm.unischema import dict_to_spark_row
+from petastorm.predicates import in_lambda
+from typing import Iterator, List
+from pathlib import Path
+
+
+class PetastormStorageEngine(AbstractStorageEngine):
+
+    def __init__(self):
+        """
+        Maintain a long live spark session and context.
+        """
+        self._spark = Session()
+        self.spark_session = self._spark.get_session()
+        self.spark_context = self._spark.get_context()
+
+    def _spark_url(self, table: DataFrameMetadata) -> str:
+        """
+        Generate a spark/petastorm url given a table
+        """
+        return Path(table.file_url).resolve().as_uri()
```

Comment:
This looks like a cleaner way to generate uris. Is the `datasets` location also set in this manner?

Reply:
Yeah, we are leveraging pathlib.Path for every system path in our system.